File: app/src/components/ScreenBackdrop.js
// src/components/ScreenBackdrop.js
// Ana ekranların arkasına çok soluk "taktik tahtası" dokusu koyan sarmalayıcı.
// Bağımlılık: react-native-svg + RN Animated (yeni paket yok). Mobil + web.

import React, { useEffect, useRef } from 'react';
import { View, Animated, StyleSheet } from 'react-native';
import Svg, {
  Defs,
  RadialGradient,
  Stop,
  Rect,
  Line,
  Circle,
  G,
} from 'react-native-svg';
import { colors } from '../theme';

const BG = colors.bg || colors.background || '#F3F5F9';
const LINE = 'rgba(11,27,58,0.09)';        // ana çizgi (soluk lacivert)
const LINE_SOFT = 'rgba(11,27,58,0.06)';   // daha soluk
const X_COLOR = 'rgba(11,27,58,0.22)';     // X işareti
const O_COLOR = 'rgba(226,27,45,0.22)';    // O işareti (accent ailesi)

function XMark({ x, y, s = 10 }) {
  return (
    <G>
      <Line x1={x - s} y1={y - s} x2={x + s} y2={y + s} stroke={X_COLOR} strokeWidth={3} strokeLinecap="round" />
      <Line x1={x - s} y1={y + s} x2={x + s} y2={y - s} stroke={X_COLOR} strokeWidth={3} strokeLinecap="round" />
    </G>
  );
}

export default function ScreenBackdrop({ children, style, animated = true }) {
  // Sadece opacity → useNativeDriver:true (web'de de sorunsuz, düşük maliyet)
  const breathe = useRef(new Animated.Value(1)).current;

  useEffect(() => {
    if (!animated) return;
    const loop = Animated.loop(
      Animated.sequence([
        Animated.timing(breathe, { toValue: 0.5, duration: 3000, useNativeDriver: true }),
        Animated.timing(breathe, { toValue: 1, duration: 3000, useNativeDriver: true }),
      ])
    );
    loop.start();
    return () => loop.stop();
  }, [animated, breathe]);

  return (
    <View style={[{ flex: 1, backgroundColor: BG }, style]}>
      {/* Statik saha çizgileri + üst parıltı */}
      <View style={StyleSheet.absoluteFill} pointerEvents="none">
        <Svg
          width="100%"
          height="100%"
          viewBox="0 0 360 720"
          preserveAspectRatio="xMidYMid slice"
        >
          <Defs>
            <RadialGradient id="topGlow" cx="50%" cy="10%" r="65%">
              <Stop offset="0%" stopColor={colors.primary || '#0B1B3A'} stopOpacity="0.10" />
              <Stop offset="100%" stopColor={colors.primary || '#0B1B3A'} stopOpacity="0" />
            </RadialGradient>
          </Defs>

          <Rect x="0" y="0" width="360" height="720" fill="url(#topGlow)" />

          {/* Saha dış çizgisi */}
          <Rect x="24" y="40" width="312" height="640" fill="none" stroke={LINE} strokeWidth="2" />
          {/* Orta çizgi + orta yuvarlak + nokta */}
          <Line x1="24" y1="360" x2="336" y2="360" stroke={LINE} strokeWidth="2" />
          <Circle cx="180" cy="360" r="52" fill="none" stroke={LINE} strokeWidth="2" />
          <Circle cx="180" cy="360" r="3" fill={LINE} />
          {/* Üst ceza sahası + kale kutusu */}
          <Rect x="92" y="40" width="176" height="92" fill="none" stroke={LINE} strokeWidth="2" />
          <Rect x="136" y="40" width="88" height="36" fill="none" stroke={LINE_SOFT} strokeWidth="2" />
          {/* Alt ceza sahası + kale kutusu */}
          <Rect x="92" y="588" width="176" height="92" fill="none" stroke={LINE} strokeWidth="2" />
          <Rect x="136" y="644" width="88" height="36" fill="none" stroke={LINE_SOFT} strokeWidth="2" />
        </Svg>
      </View>

      {/* X/O işaretleri — hafif "nefes alan" katman */}
      <Animated.View
        style={[StyleSheet.absoluteFill, { opacity: animated ? breathe : 1 }]}
        pointerEvents="none"
      >
        <Svg
          width="100%"
          height="100%"
          viewBox="0 0 360 720"
          preserveAspectRatio="xMidYMid slice"
        >
          <XMark x={100} y={250} />
          <Circle cx={250} cy={220} r={10} fill="none" stroke={O_COLOR} strokeWidth={3} />
          <XMark x={240} y={500} />
          <Circle cx={110} cy={540} r={10} fill="none" stroke={O_COLOR} strokeWidth={3} />
        </Svg>
      </Animated.View>

      {children}
    </View>
  );
}

/*
## Entegrasyon
Ana Sayfa / Bülten / Analiz / Profil ekran köklerini sarmala:
  <ScreenBackdrop>
    <ScrollView style={[styles.container, { backgroundColor: 'transparent' }]} ...>
      ...mevcut içerik...
    </ScrollView>
  </ScreenBackdrop>
FlatList/View kökleri için de aynı: kökün backgroundColor'ını 'transparent' yap
ki desen boşluklardan görünsün. Desen pointerEvents="none" — dokunuşu engellemez.
*/

File: flutter/lib/widgets/screen_backdrop.dart
// KAYNAK: app/src/components/ScreenBackdrop.js — BİREBİR çeviri.
//
// Ana ekranların arkasına çok soluk "taktik tahtası" dokusu koyan sarmalayıcı.
// Kaynakta iki SVG katmanı vardı: (1) statik saha çizgileri + üst parıltı,
// (2) 3 sn'de bir 1 ↔ 0.5 arası "nefes alan" X/O işaretleri.
//
// PRESERVEASPECTRATIO="xMidYMid slice" KARŞILIĞI: SVG'de bu, deseni ekranı
// KAPLAYACAK şekilde büyütüp ortalar (taşan kenar kırpılır). Flutter'da
// yerleşik karşılığı yok; ölçek `max(en/360, boy/720)` ile hesaplanıp tuval
// elle ortalanır. `min` kullanmak deseni ekrana SIĞDIRIR ve kenarlarda boş
// şerit bırakırdı — kaynaktaki görüntü bu değil.
//
// Desen dokunuşu ENGELLEMEZ (kaynakta pointerEvents="none"): burada
// `IgnorePointer` ile karşılanır.

import 'package:flutter/material.dart';

import '../core/theme/tokens.dart';

// DESEN ARTIK TEMAYI DİNLER (kullanıcı isteği, 2026-08-12).
//
// Kaynakta dördü de sabitti: saha çizgileri ve X marka laciverdi (11,27,58),
// O marka kırmızısı (226,27,45). Opaklıklar KAYNAKTAKİ değerlerdir, değişmedi
// — yalnız hangi rengin soluklaştığı değişti.
//
// ÇİZGİ VE X NEDEN `text`, `primary` DEĞİL: bu bir doku, vurgu değil. `text`
// zeminin karşı ucundadır (açık temada koyu, koyu temada açık), yani desen
// HER temada görünür kalır. `primary` kullansaydık Beşiktaş gibi koyu bir
// temada koyu desen koyu zeminde kaybolurdu.
//
// O NEDEN `accent`: kaynakta da ikinci renkti. Kullanıcının "iki ana renk
// dengeli kullanılsın" isteği burada da geçerli — desen tek renge inmemeli.
Color get _kLine => AppColors.text.withValues(alpha: 0.09);
Color get _kLineSoft => AppColors.text.withValues(alpha: 0.06);
Color get _kX => AppColors.text.withValues(alpha: 0.22);
Color get _kO => AppColors.accent.withValues(alpha: 0.22);

class ScreenBackdrop extends StatefulWidget {
  // `const` DEĞİL — BİLEREK (2026-08-12): bu widget rengini `AppColors`
  // küresellerinden okuyor ve tema çalışma zamanında değişiyor. `const`
  // yapıcı widget örneğini sabitler; Flutter aynı örneği görünce alt ağacı
  // YENİDEN KURMAZ ve widget eski renkte donar (emülatörde ölçüldü:
  // Dortmund temasında kupon boş-durum kartı Galatasaray bordosunda kaldı).
  // ignore: prefer_const_constructors_in_immutables
  ScreenBackdrop({super.key, required this.child, this.animated = true});

  final Widget child;
  final bool animated;

  @override
  State<ScreenBackdrop> createState() => _ScreenBackdropState();
}

class _ScreenBackdropState extends State<ScreenBackdrop>
    with SingleTickerProviderStateMixin {
  // Kaynakta sequence(3000 ms → 0.5, 3000 ms → 1) döngüsü; Flutter'da tek
  // denetleyici `reverse` ile aynı gidiş-dönüşü verir.
  late final AnimationController _c = AnimationController(
    vsync: this,
    duration: const Duration(milliseconds: 3000),
    value: 1,
    lowerBound: 0.5,
    upperBound: 1,
  );

  @override
  void initState() {
    super.initState();
    if (widget.animated) _c.repeat(reverse: true);
  }

  @override
  void dispose() {
    _c.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) => ColoredBox(
    color: AppColors.bg,
    child: Stack(
      children: [
        Positioned.fill(
          child: IgnorePointer(
            child: CustomPaint(painter: const _SahaBoyaci()),
          ),
        ),
        Positioned.fill(
          child: IgnorePointer(
            child: widget.animated
                ? AnimatedBuilder(
                    animation: _c,
                    builder: (_, _) => Opacity(
                      opacity: _c.value,
                      child: const CustomPaint(painter: _IsaretBoyaci()),
                    ),
                  )
                : const CustomPaint(painter: _IsaretBoyaci()),
          ),
        ),
        widget.child,
      ],
    ),
  );
}

/// viewBox 0 0 360 720 → tuval dönüşümü ("slice": kaplayacak kadar büyüt,
/// ortala, taşanı kırp).
void _viewBox(Canvas canvas, Size size) {
  const w = 360.0, h = 720.0;
  final k = (size.width / w) > (size.height / h)
      ? size.width / w
      : size.height / h;
  canvas.clipRect(Offset.zero & size);
  canvas.translate((size.width - w * k) / 2, (size.height - h * k) / 2);
  canvas.scale(k);
}

class _SahaBoyaci extends CustomPainter {
  const _SahaBoyaci();

  @override
  void paint(Canvas canvas, Size size) {
    canvas.save();
    _viewBox(canvas, size);

    // Üst parıltı: merkez %50/%10, yarıçap %65 — primary rengin %10'undan
    // saydama.
    final glow = Paint()
      ..shader = RadialGradient(
        center: const Alignment(0, -0.8), // cx 50%, cy 10%
        radius: 0.65 * 2, // SVG r yüzdesi çapa göre; Flutter yarıçapa göre
        colors: [
          AppColors.primary.withValues(alpha: 0.10),
          AppColors.primary.withValues(alpha: 0),
        ],
      ).createShader(const Rect.fromLTWH(0, 0, 360, 720));
    canvas.drawRect(const Rect.fromLTWH(0, 0, 360, 720), glow);

    final cizgi = Paint()
      ..color = _kLine
      ..style = PaintingStyle.stroke
      ..strokeWidth = 2;
    final soluk = Paint()
      ..color = _kLineSoft
      ..style = PaintingStyle.stroke
      ..strokeWidth = 2;

    // Saha dış çizgisi
    canvas.drawRect(const Rect.fromLTWH(24, 40, 312, 640), cizgi);
    // Orta çizgi + orta yuvarlak + nokta
    canvas.drawLine(const Offset(24, 360), const Offset(336, 360), cizgi);
    canvas.drawCircle(const Offset(180, 360), 52, cizgi);
    canvas.drawCircle(const Offset(180, 360), 3, Paint()..color = _kLine);
    // Üst ceza sahası + kale kutusu
    canvas.drawRect(const Rect.fromLTWH(92, 40, 176, 92), cizgi);
    canvas.drawRect(const Rect.fromLTWH(136, 40, 88, 36), soluk);
    // Alt ceza sahası + kale kutusu
    canvas.drawRect(const Rect.fromLTWH(92, 588, 176, 92), cizgi);
    canvas.drawRect(const Rect.fromLTWH(136, 644, 88, 36), soluk);

    canvas.restore();
  }

  @override
  bool shouldRepaint(_SahaBoyaci old) => false;
}

class _IsaretBoyaci extends CustomPainter {
  const _IsaretBoyaci();

  @override
  void paint(Canvas canvas, Size size) {
    canvas.save();
    _viewBox(canvas, size);

    final x = Paint()
      ..color = _kX
      ..style = PaintingStyle.stroke
      ..strokeWidth = 3
      ..strokeCap = StrokeCap.round;
    final o = Paint()
      ..color = _kO
      ..style = PaintingStyle.stroke
      ..strokeWidth = 3;

    void carpi(double cx, double cy, [double s = 10]) {
      canvas.drawLine(Offset(cx - s, cy - s), Offset(cx + s, cy + s), x);
      canvas.drawLine(Offset(cx - s, cy + s), Offset(cx + s, cy - s), x);
    }

    carpi(100, 250);
    canvas.drawCircle(const Offset(250, 220), 10, o);
    carpi(240, 500);
    canvas.drawCircle(const Offset(110, 540), 10, o);

    canvas.restore();
  }

  @override
  bool shouldRepaint(_IsaretBoyaci old) => false;
}
